frontend/app/src/shared/components/form/object-form.tsx

```diff
@@ -87,60 +89,116 @@ const ObjectForm = ({ kind, currentProfiles, ...props }: ObjectFormProps) => {
   if ([REPOSITORY_KIND, READONLY_REPOSITORY_KIND].includes(kind)) {
     return (
       <Suspense fallback={<LoadingIndicator className="mt-4" />}>
-        <RepositoryForm schema={schema} {...props} />
+        <RepositoryForm
+          schema={schema}
+          parentSchema={parentSchema}
+          parentData={parentData}
+          {...props}
+        />
       </Suspense>
     );
   }
 
   if (kind === NUMBER_POOL_OBJECT) {
-    return <NumberPoolForm {...props} />;
+    return <NumberPoolForm parentSchema={parentSchema} parentData={parentData} {...props} />;
```

Comment:
instead of props drilling on every child components, you can call useFromContext directly in those who need it.